[SYCL][COMPAT] Add get_error_string_dummy() function (#15719)
intel/llvm: sycl/doc/syclcompat/README.md

`syclcompat::error_code::default_error` is returned instead. For both cases, it
prints the error message to the standard error stream.

`get_error_string_dummy` is a dummy function introduced to assist auto

> migration. It needs to be replaced with a real error-handling function. SYCL
> reports errors using exceptions and does not use error codes.

Reviewer: ```suggestion
migration. The SYCLomatic user should replace it with a real error-handling 
function. SYCL reports errors using exceptions and does not use error codes.
```

Reviewer: I suggest this clarification because otherwise it could be misinterpreted as a TODO for the SYCLcompat team.

Author: Updated in 0ffef5c
